File: src/main/java/zechs/zplex/sync/data/remote/DriveApi.java
package zechs.zplex.sync.data.remote;

import com.google.api.services.drive.model.File;
import zechs.zplex.sync.data.remote.model.DrivePathFile;

import java.io.IOException;
import java.util.List;

public interface DriveApi {

    List<File> getFiles(String folderId, Boolean foldersOnly) throws IOException;

    List<DrivePathFile> getAllFilesRecursively(String folderId);

}

File: src/main/kotlin/data/remote/DriveApi.kt
package zechs.zplex.sync.data.remote

import com.google.api.services.drive.model.File
import zechs.zplex.sync.data.model.DrivePathFile

interface DriveApi {

    fun getFiles(folderId: String, foldersOnly: Boolean): List<File>
    fun getAllFilesRecursively(folderId: String): List<DrivePathFile>

}